src/dotnet/Core/Uploads/UploadProcessorsExt.cs

```diff
@@ -0,0 +1,43 @@
+using Stl.IO;
+
+namespace ActualChat.Uploads;
+
+public static class UploadProcessorsExt
+{
+    public static async Task<ProcessedFile> Process(this IEnumerable<IUploadProcessor> processors, FilePath fileName, string contentType, Stream stream, CancellationToken cancellationToken)
+    {
+        var processor = processors.FirstOrDefault(x => x.Supports(contentType));
+        if (processor is null) {
+            // no need to dump file to file system
+            var memoryFile = await ToMemoryFile(fileName, contentType, stream, cancellationToken).ConfigureAwait(false);
```

Comment:
Why do we copy file content to memory?
Can't we refer to original stream?

Reply:
done